plugins/lua/spectate.lua

```diff
@@ -91,16 +134,16 @@ local function do_toggle()
 end
 
 local function set_setting(key, value)
-    if config.data[key] == nil then
+    if config[key] == nil then
         qerror('unknown setting: ' .. key)
     end
     if key == 'follow-seconds' then
         value = argparse.positiveInt(value, 'follow-seconds')
     else
         value = argparse.boolean(value, key)
     end
-    config.data[key] = value
-    config:write()
+    config[key] = value
```

Comment:
how does this work with the table-valued settings?

Reply:
haven't gotten to that yet, I was thinking of an optional 3rd argument to `set_settings`.

Reply:
`spectate set tooltip-stress-levels 6 text XD` is now valid command